### PR TITLE
ext_proc: Improve test coverage in matching_utils

### LINKED ISSUE #34496: ext_proc: Improve the test coverage on matching_utils
From coverage report the matching_utils.h 's coverage is only 76.2% which is way below threshold.
https://storage.googleapis.com/envoy-pr/27f6f9d/coverage/source/extensions/filters/http/ext_proc/index.html
https://storage.googleapis.com/envoy-pr/27f6f9d/coverage/source/extensions/filters/http/ext_proc/matching_utils.cc.gcov.html

This issue has been hidden because other components in ext_proc have relatively high coverage.

### PR DESCRIPTION
**Pull request by jbohanon**
Commit Message: ext_proc: Improve test coverage in matching_utils
Additional Description:
Risk Level: LOW
Testing: Matching utils unit tests added, CelValue types covered more extensively in integration tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A
Fixes #34496

### CODE REVIEW

File: test/extensions/filters/http/ext_proc/matching_utils_test.cc [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#include <algorithm>
```
Comment (jbohanon) [inline test/extensions/filters/http/ext_proc/matching_utils_test.cc:1]: very much do not need all of these imports; need to figure out which are actually needed

File: source/extensions/filters/http/ext_proc/matching_utils.cc [hunk 2]
```diff
@@ -63,16 +63,35 @@ ExpressionManager::evaluateAttributes(const Filters::Common::Expr::Activation& a
...
+    // - Bytes
+    //
+    // Of these, we need to handle
+    // - bool as bool
+    // - int64, uint64 as number
+    // - everything else as string via print
+    //
+    // Handling all value types here would be graceful but is not currently
+    // testable and drives down coverage %. This is not a _great_ reason to
+    // not do it; will get feedback from reviewers.
```
Comment (mattklein123) [inline source/extensions/filters/http/ext_proc/matching_utils.cc:84]: As long as not handling has a graceful fallback which it does this seems fine to me.

**Review by mattklein123 — APPROVED** [review]

**Review by tyxia — APPROVED** [review]
Thank you very much for improving the test coverage!

Optional: You can mention https://github.com/envoyproxy/envoy/issues/34496 in the PR description so that issue will be auto-closed.

**Comment by jbohanon** [comment]
/retest

Is this a known flake in the hotrestart tsan tests? https://dev.azure.com/cncf/envoy/_build/results?buildId=173926&view=logs&j=d1f76054-8f79-554b-6f4a-11d6a63b8e00&t=33c64725-b304-5065-9740-0ce244fa1a25

**Pull request merged**